src/Interpreters/executeQuery.cpp

```diff
@@ -2045,6 +2048,26 @@ void executeQuery(
 
                 const auto & out_file = typeid_cast<const ASTLiteral &>(*ast_query_with_output->out_file).value.safeGet<std::string>();
 
+                if (context->getSettingsRef()[Setting::into_outfile_create_parent_directories])
```

Comment:
This code is executed on the server side and I don't think we should create new directories on server side. Actually I doubth there are places where `INTO OUTFILE` is executed on server side at all, so let's remove this.